xplique/attributions/global_sensitivity_analysis/sobol_estimators.py

```diff
@@ -216,19 +283,28 @@ def __call__(self, masks, outputs, nb_design):
           Total order Sobol' indices, one for each dimensions.
         """
         nb_dim = self.masks_dim(masks)
-        sampling_a, _, replication_c = self.split_abc(outputs, nb_design, nb_dim)
+        a, _, c = self.split_abc(outputs, nb_design, nb_dim)  # a:(N,), c:(D,N)
+
+        a = tf.cast(a, dtype=tf.float32)
+        c = tf.cast(c, dtype=tf.float32)
+
+        N = tf.cast(tf.shape(a)[0], a.dtype)
+        denom_cov = tf.maximum(N - 1.0, 1.0)
```

Comment:
I do not see a sign of this maximum between `N - 1` and `1` in the initial code:

```python
        nb_dim = self.masks_dim(masks)
        sampling_a, _, replication_c = self.split_abc(outputs, nb_design, nb_dim)

        mu_ac = [(1. / nb_design) * np.sum(sampling_a + replication_c[i]) / 2.
                 for i in range(nb_dim)]
        var   = [(1. / (nb_design - 1.)) * np.sum(sampling_a**2. + replication_c[i]**2.) /
                  2. - mu_ac[i]**2. for i in range(nb_dim)]

        stis = [
            1. - ((1. / nb_design) * np.sum(sampling_a * replication_c[i]) - mu_ac[i]**2.) / var[i]
            for i in range(nb_dim)
        ]

        return self.post_process(stis, masks)
```

Where does it come from?